webpack/components/AnsibleHostDetail/components/AnsibleVariableOverrides/AnsibleVariableOverrides.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Skeleton from 'react-loading-skeleton';
+
+import { useQuery } from '@apollo/client';
+import variableOverrides from '../../../../graphql/queries/variableOverrides.gql';
+import AnsibleVariableOverridesTable from './AnsibleVariableOverridesTable';
+
+import { encodeId } from '../../../../globalIdHelper';
+import { reorderVariables } from './AnsibleVariableOverridesHelper';
+import './AnsibleVariableOverrides.scss';
+
+const AnsibleVariableOverrides = ({ id, hostAttrs }) => {
+  const { loading, data, error } = useQuery(variableOverrides, {
+    variables: { hostId: id, id: encodeId('Host', id) },
+  });
```

Comment:
With the hook handling the mapping to status constants, the component looks like the following:

```js
const AnsibleVariableOverrides = ({ hostAttrs }) => {
  const { id } = hostAttrs;

  const { status, data, error } = useQueryWithStatus({ hostId: id, id: encodeId('Host', id) });

  const errorNode = (<div>{error ? error.message : ''}</div>)

  return (
    <SkeletonLoader status={status} errorNode={errorNode}>
      <div className="ansible-tab-margin">
        <AnsibleVariableOverridesTable
          variables={reorderVariables(data ? data.host.allAnsibleRoles.nodes : [])}
          hostAttrs={hostAttrs}
        />
      </div>
    </SkeletonLoader>
  );
```
So I still need to check for errors and data presence in component, but now I use additional hook wrapper and status mapping - I added more things only to get what I already had. I kept it as was for now but I am willing to change if you really think the version with wrapper is better. 

> Also, with this suggested custom hook, the data can cam be stored in foreman's redux store (like useAPI), so data manipulation can be done via selectors and memorized by reselect if needed

Yes, and it would mean having the same data in 2 places - redux store and apollo cache. That would effectively neutralize some of apollo's strong features, such as [auto-updates](https://www.apollographql.com/docs/react/data/mutations/#updating-a-single-existing-entity) after mutation, because we would need to manually update the store to match the data from apollo.

I am strongly against putting any data fetched by apollo into redux store.

Reply:
>  I am strongly against putting any data fetched by apollo into redux store.

I read about this a bit further  and I agree with you, we should use apollo as a [redux replacement](https://www.apollographql.com/blog/apollo-client/caching/dispatch-this-using-apollo-client-3-as-a-state-management-solution/)